app/scripts/modules/titus/src/pipeline/stages/runJob/TitusRunJobStageConfig.tsx

```diff
@@ -0,0 +1,383 @@
+import * as React from 'react';
+import { $q } from 'ngimport';
+import { set } from 'lodash';
+
+import {
+  AccountTag,
+  IStageConfigProps,
+  NgReact,
+  RegionSelectField,
+  StageConfigField,
+  HelpField,
+  IAggregatedAccounts,
+  IRegion,
+  AccountService,
+  FirewallLabels,
+  MapEditor,
+} from '@spinnaker/core';
+
+import { DockerImageAndTagSelector, IDockerImageAndTagChanges } from '@spinnaker/docker';
+
+import { TitusSecurityGroupPicker } from './TitusSecurityGroupPicker';
+
+export interface ITitusRunJobStageConfigState {
+  credentials: string[];
+  regions: IRegion[];
+  loaded: boolean;
+}
+
+export class TitusRunJobStageConfig extends React.Component<IStageConfigProps, ITitusRunJobStageConfigState> {
+  private credentialsKeyedByAccount: IAggregatedAccounts = {};
+
+  public state: ITitusRunJobStageConfigState = {
+    credentials: [],
+    regions: [],
+    loaded: false,
+  };
+
+  private setRegistry(account: string) {
+    if (account) {
+      this.props.stage.registry = this.credentialsKeyedByAccount[account].registry;
+    }
+  }
+
+  private updateRegions(account: string) {
+    let regions: IRegion[];
+    if (account) {
+      regions = this.credentialsKeyedByAccount[account].regions;
+      if (regions.map(r => r.name).every(r => r !== this.props.stage.cluster.region)) {
+        delete this.props.stage.cluster.region;
+        this.props.stageFieldUpdated();
+        this.setState({});
```

Comment:
probs unnecessary since setState is called below with `regions`

Also I think I prefer `this.forceUpdate()` over `this.setState({})` because it's more descriptive.  What do you think?

Reply:
👍